sled-agent/src/instance.rs

```diff
@@ -518,7 +724,7 @@ impl InstanceInner {
 
             // Ensure the instance is removed from the instance manager's table
             // so that a new instance can take its place.
-            self.instance_ticket.terminate();
+            self.instance_ticket.deregister();
```

Comment:
Too many things called "terminate", so I renamed the de-registration from the `InstanceManager` BTreeMap to "deregister".